### PR TITLE
Fixed Adam initialisation

### PR DESCRIPTION
**Pull request by wdoppenberg**
### Checklist

- [X] Confirm that `run-checks.sh` has been executed.

### Related Issues/PRs

https://github.com/burn-rs/burn/issues/614
https://github.com/burn-rs/burn/pull/566 <-- would suggest to merge this first

### Changes

Changed the Adam test to have higher precision. See the related issue for more details. Updated `AdaptiveMomentumState` initialisation  (inside `transform`) to have $t_0 = 0$.

### Testing

See related issue

### CODE REVIEW

**Review by nathanielsimard — COMMENTED** [review]
@wdoppenberg Could you merge main, the github diff is quite hard to follow 😅.

**Comment by wdoppenberg** [comment]
> @wdoppenberg Could you merge main, the github diff is quite hard to follow 😅.

Apologies, will do so now
